src/containers/alerts.jsx

```diff
@@ -16,10 +16,7 @@ const Alerts = ({
 }) => (
     <AlertsComponent
         // only display standard and extension alerts here
-        alertsList={alertsList.filter(curAlert => (
-            curAlert.alertType === AlertTypes.STANDARD ||
-            curAlert.alertType === AlertTypes.EXTENSION
-        ))}
+        alertsList={filterPopupAlerts(alertsList)}
```

Comment:
Previously I suggested using a selector in `mapStateToProps`, to which you replied

> `mapStateToProps` -- the problem (as I understand it) is that if this were in mapStateToProps, it would create a new object every time redux state changes, anywhere, and thus rerender on any redux state change. Whereas like this, the props don't change unless the actual content changes, and thus it only rerenders if the actual content changes.

The solution to this is to memoize the selector:
```js
let _filterPopupAlertsArg, _filterPopupAlertsResult;
const filterPopupAlerts = alertsList => {
    if (alertsList === _filterPopupAlertsArg) return _filterPopupAlertsResult;
    _filterPopupAlertsArg = alertsList;
    _filterPopupAlertsResult = alertsList.filter(curAlert => (
        curAlert.alertType === AlertTypes.STANDARD ||
        curAlert.alertType === AlertTypes.EXTENSION
    ));
    return _filterPopupAlertsResult;
};
```
There are libraries to do this for you, `memoizee` (above is `memoize(filterPopupAlerts, {max: 1})`) and `reselect`.  However, I don't think we should add libraries unnecessarily right now, and this is readable as-is so this is just FYI.

Reply:
Thanks -- but isn't my current approach nearly, or entirely as effective? I'm calling the selector from the component's render code, not from `mapStateToProps`. So redux should only think `mapStateToProps` has produced a change if the `alertsList` reference changes, which it only will when there's actually a change we want to re-render.

Your code above just checks if if the old and new list references are the same -- that's the same thing that redux already does, right?

(I realize I have only a shaky understanding of what's going on under the hood!)

Reply:
Your approach has the same effectiveness performance-wise, but organizationally it puts filtering logic into the component, when the filtering could happen one level up. That would make the component less surprising since it would always output what you gave it.

You rightly pointed out that doing that filtering one level up in `mapStateToProps` would incur a performance hit, but that is why the selector-generating library `reselect` does memoization for you, so you don't have to worry about the details of performance to organize the code this way.

Reply:
Thanks, that makes sense. I made this follow on: https://github.com/LLK/scratch-gui/issues/3909